src/navigation/initialRouteNameState/useInitialRouteNameStateContext.ts

```diff
@@ -0,0 +1,30 @@
+import React, { useContext, useEffect, useRef } from 'react';
```

Comment:
[Q] Not critical but putting it here for discussion, will maybe a custom useEffect hook a better choice for this solution? Provided that we can remove the need for prevInitialRouteName value, the useEffect hook should allow us to run any side effect without persisting the state